### PR TITLE
Minor Tweaks to lightning-invoice for C bindings

### PR DESCRIPTION
**Pull request by TheBlueMatt**
This + https://github.com/lightningdevkit/ldk-c-bindings/pull/19 correctly builds the new invoice construction utility method for C (+Java).

### CODE REVIEW

File: lightning-invoice/src/ser.rs [hunk 1]
```diff
@@ -122,6 +123,7 @@ impl Display for SignedRawInvoice {
 	}
 }
 
+/// (C-not exported)
```
Comment (jkczyz) [inline lightning-invoice/src/ser.rs:126]: Looks unrelated to this commit.

**Review by jkczyz — APPROVED** [review]

**Review by ariard — COMMENTED** [review]
Code Review ACK 54c97ee

**Comment by TheBlueMatt** [comment]
Squashed one fixup commit without diff. Will merge after CI
```
$ git diff-tree -U3 54c97ee3 83ab933f
$
```

**Review by arik-so — APPROVED** [review]
